livre/lib/data/BoundingAxis.cpp

```diff
@@ -0,0 +1,374 @@
+}
+
+BoundingAxis::~BoundingAxis()
+{
+    _cleanGPUMemory();
+}
+
+void BoundingAxis::_sendDataToGPU( const Floats& vertices, const Floats& normals,
+                                   const Floats& normals2, const Floats& colors,
+                                   const UInt8s& types )
+{
+    glGenBuffers( 1, &_vertexeBufferId );
+    glBindBuffer( GL_ARRAY_BUFFER, _vertexeBufferId );
+    glBufferData( GL_ARRAY_BUFFER, sizeof(float) * vertices.size(),
+                  &vertices[0], GL_STATIC_DRAW );
+
+    glGenBuffers( 1, &_normalBufferId );
+    glBindBuffer( GL_ARRAY_BUFFER, _normalBufferId );
+    glBufferData( GL_ARRAY_BUFFER, sizeof(GLuint) * normals.size(),
+                  &normals[0], GL_STATIC_DRAW );
+
+    glGenBuffers( 1, &_normalBufferId2 );
+    glBindBuffer( GL_ARRAY_BUFFER, _normalBufferId2 );
+    glBufferData( GL_ARRAY_BUFFER, sizeof(GLuint) * normals2.size(),
+                  &normals2[0], GL_STATIC_DRAW );
+
+    glGenBuffers( 1, &_colorBufferId );
+    glBindBuffer( GL_ARRAY_BUFFER, _colorBufferId );
+    glBufferData( GL_ARRAY_BUFFER, sizeof(GLuint) * colors.size(),
+                  &colors[0], GL_STATIC_DRAW );
+
+    glGenBuffers( 1, &_typeBufferId );
+    glBindBuffer( GL_ARRAY_BUFFER, _typeBufferId );
+    glBufferData( GL_ARRAY_BUFFER, sizeof(GLubyte) * types.size(),
+                  &types[0], GL_STATIC_DRAW );
+}
+
+void BoundingAxis::draw()
+{
+    glEnableVertexAttribArray( 0 );
+    glBindBuffer( GL_ARRAY_BUFFER, _vertexeBufferId );
+    glVertexAttribPointer( 0, 3, GL_FLOAT, GL_FALSE, 0, 0 );
+
+    glEnableVertexAttribArray( 1 );
+    glBindBuffer( GL_ARRAY_BUFFER, _normalBufferId );
+    glVertexAttribPointer( 1, 3, GL_FLOAT, GL_FALSE, 0, 0 );
+
+    glEnableVertexAttribArray( 2 );
+    glBindBuffer( GL_ARRAY_BUFFER, _normalBufferId2 );
+    glVertexAttribPointer( 2, 3, GL_FLOAT, GL_FALSE, 0, 0 );
+
+    glEnableVertexAttribArray( 3 );
+    glBindBuffer( GL_ARRAY_BUFFER, _colorBufferId );
+    glVertexAttribPointer( 3, 4, GL_FLOAT, GL_FALSE, 0, 0 );
+
+    glEnableVertexAttribArray( 4 );
+    glBindBuffer( GL_ARRAY_BUFFER, _typeBufferId );
+    glVertexAttribIPointer( 4, 1, GL_UNSIGNED_BYTE, 0, 0 );
+
+    glDrawArrays( GL_LINES, 0, nVertices );
+
+    glDisableVertexAttribArray( 0 );
+    glDisableVertexAttribArray( 1 );
+    glDisableVertexAttribArray( 2 );
+    glDisableVertexAttribArray( 3 );
+}
+
+void BoundingAxis::_cleanGPUMemory()
+{
+    glDeleteBuffers( 1, &_vertexeBufferId );
+    glDeleteBuffers( 1, &_normalBufferId );
+    glDeleteBuffers( 1, &_normalBufferId2 );
+    glDeleteBuffers( 1, &_colorBufferId );
+    glDeleteBuffers( 1, &_typeBufferId );
+}
+
+void BoundingAxis::_createTicks( const VolumeInformation& volInfo, Floats& vertices,
+                                 Floats& normals, Floats& normals2, Floats& colors, UInt8s& types )
+{
+    const size_t longestDimCoordIndex = volInfo.worldSize.find_max_index();
```

Comment:
This is the index ( 1 for x, 2 for y, 3 for z ) of the longest dimension of the bounding box or longest axis... longestAxis is not very informative ?